src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -526,10 +534,14 @@ public void setDatasetBatchSize(Integer datasetBatchSize) {
     this.datasetBatchSize = datasetBatchSize;
   }
 
-  public Dataset updateDatasetIndex(Integer datasetId, Integer userId, Instant indexDate)
-      throws SQLException {
-    datasetServiceDAO.updateDatasetIndex(datasetId, userId, indexDate);
-    return datasetDAO.findDatasetById(datasetId);
+  protected void updateIfIndexed(Dataset dataset, Integer userId, Instant indexDate) {
```

Comment:
Every caller to this method passes `Instant.now()`. Instead you can omit the argument and use `Instant.now()` directly in the code. It looks like all other caller to `datasetServiceDAO.updateDatasetIndex()` also pass in `Instant.now()` except for the case where it's set to `null`. You could split the DAO layer into two methods, update and delete and get rid of the argument completely.

Reply:
Furthermore, I don't see why the date is stored at all. The only use is to compare it against `null`, so it's being treated like a boolean flag. Is there a future use case where the index date would be useful?

Although, maybe this should be part of a more general approach to indexing.

Reply:
>  Is there a future use case where the index date would be useful?

I think the potential use case is not necessarily for traditional user's to see, but for admins to see when it was last indexed separate from the most recent update date, which is something we also store, but is not modified by the indexing operations.